### PR TITLE
Conditionally import `...names()`

### LINKED ISSUE #241: oe_get - Error in ...names() : could not find function "...names"
Hi! 
I want to use OSM data using R and I found this wonderful package `osmextract`. But when I run a simple command to download all polylines available in OSM for India, I get this error shown below. Any help is appreciated, thanks! (This error comes up even when I try to use the examples shown in the [README ](https://github.com/ropensci/osmextract)of `osmextract`). I saw [this](https://github.com/ropensci/osmextract/issues/234), but sorry could not understand much. 

```{r}
library(sf)
library(osmextract)
oe_get(place = "India", layer = "lines", provider = "geofabrik", match_by = "name") 
# The input place was matched with: India
# Error in ...names() : could not find function "...names"
```

### PR DESCRIPTION
**Pull request by agila5**
Fix #241. 

The problem is that the `...names()` function was introduced in `R` v4.1 while we condition only on `R` >= 3.5. It is quite difficult to reimplement the existing code without `...names()`, so I decided to add a new dependency (`backports`) and conditionally import `...names()` when R < 4.1.

### CODE REVIEW

**Comment by agila5** [comment]
Unfortunately, we need to wait for https://github.com/r-lib/backports/issues/65 (or, maybe, I will just remove those tests)

**Comment by agila5** [comment]
Unfortunately that still doesn't work, see the issue in the backports repo...

**Comment by agila5** [comment]
At the end, I reimplemented the same approach using `names(list(...))` and `tryCatch` (i.e. without `backports` or `...names()`).

File: R/get-key-values.R [hunk 1]
```diff
@@ -62,7 +61,7 @@
 #' @examples
 #' # Copy ITS file to tempdir so that the examples do not require internet
 #' # connection. You can skip the next few lines (and start directly with
-#' # oe_get_keys) when running the examples ocally.
+#' # oe_get_keys) when running the examples locally.
```
Comment (Robinlovelace) [inline R/get-key-values.R:65]: :+1:

File: R/get-key-values.R [hunk 2]
```diff
@@ -105,13 +104,25 @@
 #' # Remove .pbf and .gpkg files in tempdir
 #' # (since they may interact with other examples)
 #' file.remove(list.files(path = tempdir(), pattern = "(pbf|gpkg)", full.names = TRUE))
-oe_get_keys = function(zone, layer = "lines", values = FALSE, which_keys = NULL) {
+oe_get_keys = function(
+  zone,
```
Comment (Robinlovelace) [inline R/get-key-values.R:109]: Really like this style, with each argument on a new line.

File: R/get-key-values.R [hunk 3]
```diff
@@ -191,28 +208,25 @@ oe_get_keys.character = function(zone, layer = "lines", values = FALSE, which_ke
...
-      quiet = TRUE
-    )
-  } else {
-    obj = sf::st_read(
-      dsn = zone,
-      query = paste0("select other_tags from ", layer),
-      quiet = TRUE
-    )
-  }
+  obj = my_st_read(
```
Comment (Robinlovelace) [inline R/get-key-values.R:212]: :eyes:

Comment (Robinlovelace) [reply R/get-key-values.R:212]: Looking at this :+1: 

https://github.com/ropensci/osmextract/blob/remove-.names/R/utils.R#L31-L57

**Review by Robinlovelace — APPROVED** [review]

**Comment by Robinlovelace** [comment]
Cannot see the new dep on backports and TBH not 100% sure what is going on. My general thought is that there's no harm in insisting on recent R versions. But if it works and enables more people to use the package, as seems to be the case, :+1: from me, as per approval in review.

**Pull request merged**